packages/components/src/components/hds/copy/button/index.ts

```diff
@@ -15,9 +15,27 @@ export const SUCCESS_ICON = 'clipboard-checked';
 export const ERROR_ICON = 'clipboard-x';
 export const DEFAULT_STATUS = 'idle';
 
-export default class HdsCopyButtonComponent extends Component {
+export interface HdsCopyButtonSignature {
+  Args: {
+    isFullWidth?: boolean;
+    isIconOnly?: boolean;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onError?: (...args: any[]) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onSuccess?: (...args: any[]) => void;
+    size?: 'small' | 'medium'; // TODO: create new type?
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    targetToCopy?: string | ((...args: any[]) => void); // TODO: create new type?
+    text: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    textToCopy?: string | ((...args: any[]) => void); // TODO: create new type?
```

Comment:
For the above TODOs should I be adding a `types.ts` file and define new types for them?

Reply:
Also should I be using different types than `any`?

Reply:
For consistency, yes I would create an enum for size and, indeed you can define a type that can be shared by targetToCopy and textToCopy (as I understand they both share the same potential structure)

Reply:
Pushed up